kne_cli/main.go

```diff
@@ -40,9 +45,50 @@ func main() {
 		}
 	}
 	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
-	if err := cmd.ExecuteContext(context.Background()); err != nil {
-		klog.Flush()
+	err := cmd.ExecuteContext(context.Background())
+	// err is referenced at the end of this function.
+	klog.Flush()
+
+	// klog provides no mechanism to determine where the logs are.  This
+	// code mimics the decsion that klog makes to determine the logging
+	// directory.
+
+	f := flag.Lookup("log_dir")
```

Comment:
Thoughts on having another func `printLogFilenames` or similar to contain lines 57-89?

then the flow is a little more straightforward with the err val on line 48 being evaluated directly in line instead of needing the comment explaining that it is evaluated later

Not a big deal either way, code LGTM